### PR TITLE
Refactors flashes to ECS, sunglasses protect you from being flashed.

### PR DESCRIPTION
**Pull request by Zumorica**
## About the PR 
- Refactors Flashable and Flash to be entirely ECS
- Adds a `FlashAttemptEvent` which can be cancelled to prevent a mob from being flashed.
- Adds FlashImmunityComponent, which will always cancel the event above unless disabled.
- Sunglasses now protect you from being flashed!
- Change shitty glasses description.

**Changelog**

:cl:
- tweak: Sunglasses now protect you against flashes.

### CODE REVIEW

File: Content.Server/Flash/FlashSystem.cs [hunk 1]
```diff
@@ -63,7 +76,7 @@ public void OnUseInHand(EntityUid uid, FlashComponent comp, UseInHandEvent args)
 
             foreach (var entity in IoCManager.Resolve<IEntityLookup>().GetEntitiesInRange(comp.Owner.Transform.Coordinates, comp.Range))
             {
-                FlashEntity(entity, args.User, comp.AoeFlashDuration, comp.SlowTo);
+                Flash(entity.Uid, args.User.Uid, uid, comp.AoeFlashDuration, comp.SlowTo);
```
Comment (metalgearsloth) [inline Content.Server/Flash/FlashSystem.cs:79]: Github doesn't let me select the line but this iterator is doing the resolve for IEntityLookup when it's already cached.

File: Content.Server/Flash/Components/FlashImmunityComponent.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Content.Server.Flash.Components
+{
+    [RegisterComponent, Friend(typeof(FlashSystem))]
+    public class FlashImmunityComponent : Component
+    {
+        public override string Name => "FlashImmunity";
+
+        [DataField("enabled")]
+        public bool Enabled { get; set; } = true;
```
Comment (metalgearsloth) [inline Content.Server/Flash/Components/FlashImmunityComponent.cs:13]: VV too IMO.

File: Content.Server/Flash/FlashSystem.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Server/Flash/FlashSystem.cs:78]: Seems I forgot this one

File: Content.Server/Flash/Components/FlashImmunityComponent.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/Flash/Components/FlashImmunityComponent.cs]: Done

**Comment by Zumorica** [comment]
Addressed reviews and the tiny nitpicks you brought up on discord before

**Pull request merged**